STCore/src/commonMain/kotlin/com/infomaniak/multiplatform_swisstransfer/managers/UploadTokensManager.kt

```diff
@@ -0,0 +1,122 @@
+     * @throws CancellationException If the operation is cancelled.
+     */
+    @Throws(RealmException::class, CancellationException::class)
+    suspend fun getTokenForEmail(email: String): String? = withContext(Dispatchers.Default) {
+        return@withContext uploadTokensController.getEmailTokenForEmail(email)?.token
+    }
+
+    /**
+     * Asynchronously sets the email and it's corresponding token.
+     *
+     * @param email The validated email.
+     * @param emailToken The valid token associated to the email.
+     *
+     * @throws RealmException If an error occurs during database access.
+     * @throws CancellationException If the operation is cancelled.
+     */
+    @Throws(RealmException::class, CancellationException::class)
+    suspend fun setEmailToken(email: String, emailToken: String): Unit = withContext(Dispatchers.Default) {
+        uploadTokensController.setEmailToken(email, emailToken)
+    }
+    //endregion
+
+    //region Attestation token
+
+    /**
+     * Get the current attestation token.
+     *
+     * @return A token which is used to attest the app integrity to the API.
+     *
+     * @throws RealmException If an error occurs during database access.
+     * @throws CancellationException If the operation is cancelled.
+     * @throws InvalidAttestationTokenException If the token is expired or if we couldn't get a new one from the API
+     */
+    @Throws(RealmException::class, CancellationException::class, InvalidAttestationTokenException::class)
+    suspend fun getAttestationToken(): String = withContext(Dispatchers.Default) {
+        return@withContext uploadTokensController.getAttestationToken()?.token?.also(::assertAttestationTokenValidity)
+            ?: throw InvalidAttestationTokenException("Missing token")
+    }
+
+    /**
+     * Asynchronously sets the attestation token.
+     *
+     * @param attestationToken The valid attestation token used to attest the app integrity to the API.
+     *
+     * @throws RealmException If an error occurs during database access.
+     * @throws CancellationException If the operation is cancelled.
+     */
+    @Throws(RealmException::class, CancellationException::class)
+    suspend fun setAttestationToken(attestationToken: String): Unit = withContext(Dispatchers.Default) {
+        uploadTokensController.setAttestationToken(attestationToken)
+    }
+
+    private fun assertAttestationTokenValidity(attestationToken: String) {
+        val tokenExpiryAt = runCatching {
+            decodeJwtToken(attestationToken)!!
+        }.getOrElse {
+            throw InvalidAttestationTokenException("Error decoding token", cause = it)
+        }
+
+        if (tokenExpiryAt < Clock.System.now().epochSeconds) {
+            throw InvalidAttestationTokenException("Local token is expired")
+        }
+    }
+
+    @OptIn(ExperimentalEncodingApi::class)
+    private fun decodeJwtToken(token: String): Long? {
+        val encodedPayload = token.split('.')[1]
```

Comment:
Will it crash if the token has a bad format ? Can we make it safer ?

Reply:
I'll also add tests for this method in a next PR